homeassistant/components/eq3btsmart/entity.py

```diff
@@ -35,92 +25,24 @@ def __init__(
         suffix = f"_{unique_id_key}" if unique_id_key else ""
-        self._attr_unique_id = f"{format_mac(self._eq3_config.mac_address)}{suffix}"
+        self._attr_unique_id = f"{format_mac(self._mac_address)}{suffix}"
 
     async def async_added_to_hass(self) -> None:
         """Run when entity about to be added to hass."""
-
-        self._thermostat.register_callback(
-            Eq3Event.DEVICE_DATA_RECEIVED, self._async_on_device_updated
-        )
-        self._thermostat.register_callback(
-            Eq3Event.STATUS_RECEIVED, self._async_on_status_updated
-        )
-        self._thermostat.register_callback(
-            Eq3Event.SCHEDULE_RECEIVED, self._async_on_status_updated
-        )
-
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass,
-                f"{SIGNAL_THERMOSTAT_DISCONNECTED}_{self._eq3_config.mac_address}",
-                self._async_on_disconnected,
-            )
-        )
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass,
-                f"{SIGNAL_THERMOSTAT_CONNECTED}_{self._eq3_config.mac_address}",
-                self._async_on_connected,
-            )
-        )
-
-    async def async_will_remove_from_hass(self) -> None:
-        """Run when entity will be removed from hass."""
-
-        self._thermostat.unregister_callback(
-            Eq3Event.DEVICE_DATA_RECEIVED, self._async_on_device_updated
-        )
-        self._thermostat.unregister_callback(
-            Eq3Event.STATUS_RECEIVED, self._async_on_status_updated
-        )
-        self._thermostat.unregister_callback(
-            Eq3Event.SCHEDULE_RECEIVED, self._async_on_status_updated
-        )
-
-    @callback
-    def _async_on_status_updated(self, data: Any) -> None:
-        """Handle updated status from the thermostat."""
-
-        self.async_write_ha_state()
-
-    @callback
-    def _async_on_device_updated(self, data: Any) -> None:
-        """Handle updated device data from the thermostat."""
-
-        self.async_write_ha_state()
+        await super().async_added_to_hass()
 
     @callback
-    def _async_on_disconnected(self) -> None:
-        """Handle disconnection from the thermostat."""
-
-        self._attr_available = False
-        self.async_write_ha_state()
-
-    @callback
-    def _async_on_connected(self) -> None:
-        """Handle connection to the thermostat."""
-
-        self._attr_available = True
-        self.async_write_ha_state()
-
-    @property
-    def available(self) -> bool:
-        """Whether the entity is available."""
-
-        try:
-            _ = self._thermostat.status
-        except Eq3Exception:
-            return False
-
-        return self._attr_available
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_available = self.coordinator.last_update_success
+        super()._handle_coordinator_update()
```

Comment:
CoordinatorEntity doesn't incoroporate `_attr_available` so this callback isn't doing anything as the behaviour you tried to add already was the default for the coordinator entity